File: Tree/BT_traversal.cpp
/*
Tree : visited array 대신 parent array && depth array 사용
중위순회(Inorder Traversal) : 왼쪽 서브트리를 중위순회 -> 현재 정점 방문 -> 오른쪽 서브트리 중위순회 
*/
#include <bits/stdc++.h>
using namespace std;

int lc[9] = {0,2,4,6,0,0,0,0,0};
int rc[9] = {0,3,5,7,0,8,0,0,0};

void preorder(int cur){
    cout << cur << ' ';
    if (lc[cur] != 0) preorder(lc[cur]);
    if (rc[cur] != 0) preorder(rc[cur]);
}


void inorder(int cur){
    if (lc[cur] != 0) preorder(lc[cur]);
    cout << cur << ' ';
    if (rc[cur] != 0) preorder(rc[cur]);
}

int main(){
    preorder(1);
    inorder(1);
}

File: Tree/BT_traversal.py
lc = [0, 2, 4, 6, 0, 0, 0, 0, 0]
rc = [0, 3, 5, 7, 0, 8, 0, 0, 0]

def preorder(curr):
    if (curr == 0):
        return
    print(curr, end = " ")
    preorder(lc[curr])
    preorder(rc[curr])
    
preorder(1)

def inorder(curr):
    if (curr == 0):
        return 
    inorder(lc[curr])
    print(curr, end = " ")
    inorder(rc[curr])

print()
inorder(1)

def postorder(curr):
    if (curr == 0):
        return 
    postorder(lc[curr])
    postorder(rc[curr])
    print(curr, end = " ")

print()
postorder(1)
